src/app/asgi.py

```diff
@@ -51,4 +64,18 @@ def _base_model_encoder(value: BaseModel) -> dict[str, Any]:
         on_app_init=[domain.security.auth.on_app_init],
         static_files_config=static_files.config,
         template_config=template_config,  # type: ignore[arg-type]
+        signature_namespace={
+            "AsyncSession": AsyncSession,
+            "FilterTypes": FilterTypes,
+            "BeforeAfter": BeforeAfter,
+            "CollectionFilter": CollectionFilter,
+            "LimitOffset": LimitOffset,
+            "UUID": UUID,
+            "EmailStr": EmailStr,
+            "datetime": datetime,
+            "User": User,
+            "ASGIConnection": ASGIConnection,
+            "Request": Request,
+            "OAuth2Login": OAuth2Login,
+        },
```

Comment:
I'm considering having the new SQLAlchemy plugin register names like `AsyncSession` and `AsyncEngine` in the sig namespace automatically - wondering if it would be worth doing something similar for the repository too. You could do the same with your `OAuth2Login` in `domain.security.auth.on_app_init`.

`Request` shouldn't need to be added as it is included by default when we parse the sigs: https://github.com/starlite-api/starlite/blob/36f337a574e17d9ac4c88f7d6869101ec3a64b2f/starlite/_signature/utils.py#L22-L33

Should I add `ASGIConnection` to those?

Reply:
Let me go back and review the code.  I may have proactively added `ASGIConnection` and it's not actually needed.

There are likely going to be others besides the `OAuth2Login` so it would be worthwhile to do something like this in the security components.  I'll take a look at your implementation on this and follow your pattern.